plugins/in_winevtlog/winevtlog.c

```diff
@@ -0,0 +1,786 @@
+
+    return result;
+}
+
+static void close_handles(struct winevtlog_channel *ch)
+{
+    if (ch->subscription) {
+        EvtClose(ch->subscription);
+        ch->subscription = NULL;
+    }
+    if (ch->signal_event) {
+        CloseHandle(ch->signal_event);
+        ch->signal_event = NULL;
+    }
+    if (ch->bookmark) {
+        EvtClose(ch->bookmark);
+        ch->bookmark = NULL;
+    }
+    for (int i = 0; i < ch->count; i++) {
+        if (ch->events[i]) {
+            EvtClose(ch->events[i]);
+            ch->events[i] = NULL;
+        }
+    }
+    ch->count = 0;
+}
+
+
+void winevtlog_close(struct winevtlog_channel *ch)
+{
+    flb_free(ch->name);
+    close_handles(ch);
+
+    flb_free(ch);
+}
+
+// Render the event as an XML string and print it.
+PWSTR render_event(EVT_HANDLE hEvent, DWORD flags, unsigned int *event_size)
+{
+    DWORD status = ERROR_SUCCESS;
+    DWORD dwBufferSize = 0;
+    DWORD dwBufferUsed = 0;
+    DWORD count = 0;
+    LPWSTR wEventXML = NULL;
+
+    if (flags != EvtRenderEventXml && flags != EvtRenderBookmark) {
+        flb_error("Invalid flags is specified: %d", flags);
+        return NULL;
+    }
+
+    if (!EvtRender(NULL, hEvent, flags, dwBufferSize, wEventXML, &dwBufferUsed, &count))
+    {
+        if (ERROR_INSUFFICIENT_BUFFER == (status = GetLastError()))
+        {
+            dwBufferSize = dwBufferUsed;
+            /* return buffer size */
+            *event_size = dwBufferSize;
+            wEventXML = (LPWSTR)flb_malloc(dwBufferSize);
+            if (wEventXML)
+            {
+                EvtRender(NULL, hEvent, flags, dwBufferSize, wEventXML, &dwBufferUsed, &count);
+            }
+            else
+            {
+                flb_error("malloc failed");
+                goto cleanup;
+            }
+        }
+
+        if (ERROR_SUCCESS != (status = GetLastError()))
+        {
+            flb_error("EvtRender failed with %d", GetLastError());
+            goto cleanup;
+        }
+    }
+
+    return wEventXML;
+
+cleanup:
+
+    if (wEventXML)
+        flb_free(wEventXML);
+
+    return NULL;
+}
+
+DWORD render_system_event(EVT_HANDLE hEvent, PEVT_VARIANT *pSystem, unsigned int *system_size)
+{
+    DWORD status = ERROR_SUCCESS;
+    EVT_HANDLE hContext = NULL;
+    DWORD dwBufferSize = 0;
+    DWORD dwBufferUsed = 0;
+    DWORD dwPropertyCount = 0;
+    PEVT_VARIANT pValues = NULL;
+
+    hContext = EvtCreateRenderContext(0, NULL, EvtRenderContextSystem);
+    if (NULL == hContext) {
+        status = GetLastError();
+        flb_error("failed to create RenderContext with %d", status);
+
+        goto cleanup;
+    }
+    if (!EvtRender(hContext,
+                   hEvent,
+                   EvtRenderEventValues,
+                   dwBufferSize,
+                   pValues,
+                   &dwBufferUsed,
+                   &dwPropertyCount)) {
+        status = GetLastError();
+
+        if (ERROR_INSUFFICIENT_BUFFER == status) {
```

Comment:
Could you invert the terms in this comparison?

Reply:
Fixed in https://github.com/fluent/fluent-bit/pull/4179/commits/a6583c0c04c813dc128074f8223dd90a3777ce78